src/frontend/screens/Observation/index.tsx

```diff
@@ -51,12 +49,20 @@ const ObservationNotFound = () => (
   </CenteredView>
 );
 
-type Props = {
-  navigation: NavigationProp,
-};
+const Observation: NativeNavigationComponent<"Observation"> = ({
+  route,
+  navigation,
+}) => {
+  const { observationId } = route.params;
+
+  React.useLayoutEffect(() => {
+    navigation.setOptions({
+      headerRight: () => (
+        <ObservationHeaderRight observationId={observationId} />
+      ),
+    });
+  }, [navigation, observationId, ObservationHeaderRight]);
```

Comment:
```suggestion
  }, [navigation, observationId]);
```

Unnecessary dep (out of context variable)